docs/import/add_disaster.js

```diff
@@ -0,0 +1,289 @@
+  }
+  if (isNaN(year)) {
+    setStatus('Error: Year must be a number.');
+    return Promise.resolve(false);
+  }
+  if (notAllLowercase(name)) {
+    setStatus(
+        'Error: disaster name must be comprised of only lowercase letters');
+    return Promise.resolve(false);
+  }
+  const disasterId = year + '-' + name;
+  return writeNewDisaster(disasterId, states);
+}
+
+/**
+ * Returns true if the given string is *not* all lowercase letters.
+ * @param {string} val
+ * @return {boolean}
+ */
+function notAllLowercase(val) {
+  return !/^[a-z]+$/.test(val);
+}
+
+// Needed for testing :/
+const emptyCallback = () => {};
+
+// TODO: add functionality for adding assets to disaster records from these
+// pickers.
+/**
+ * Requests all assets in ee directories corresponding to given states.
+ * @param {Array<string>} states e.g. ['WA']
+ * @return {Promise<Array<string | Array<string>>>} 2-d array of all retrieved
+ * assets in the form [['WA', ['asset/path']], ...]
+ */
+function getAssetsFromEe(states) {
+  return ee.data.listAssets(legacyStateDir, {}, emptyCallback)
+      .then((result) => {
+        const folders = new Set();
+        for (const folder of result.assets) {
+          folders.add(folder.id.substring(eeStatePrefixLength));
+        }
+        const promises = [];
+        for (const state of states) {
+          const dir = legacyStateDir + '/' + state;
+          if (!folders.has(state)) {
+            ee.data.createFolder(dir, false, () => {
+              ee.data.setAssetAcl(dir, {all_users_can_read: true});
+            });
+            promises.push(Promise.resolve([state, []]));
+          } else {
+            promises.push(
+                ee.data.listAssets(dir, {}, emptyCallback).then((result) => {
+                  const assets = [];
+                  if (result.assets) {
+                    for (const asset of result.assets) {
+                      if (checkSupportedAssetType(asset)) {
+                        assets.push(asset.id);
+                      }
+                    }
+                  }
+                  return [state, assets];
+                }));
+          }
+        }
+        return Promise.all(promises);
+      });
+}
+
+// TODO: surface a warning if unsupported asset types are found?
+/**
+ * Check that the type of the given asset is one we support (Unsupported:
+ * ALGORITHM, FOLDER, UNKNOWN).
+ * @param {Object} asset
+ * @return {boolean}
+ */
+function checkSupportedAssetType(asset) {
+  const type = asset.type;
+  return type === 'IMAGE' || type === 'IMAGE_COLLECTION' || type === 'TABLE';
+}
+
+/**
+ * Given states, displays their assets in pickers. Right now, selecting on
+ * those pickers doesn't actually do anything.
+ * @param {Array<string>} states e.g. ['WA']
+ */
+function createAssetPickers(states) {
+  const assetPickerDiv = $('#asset-pickers');
+  for (const state of states) {
+    const assetPicker = $(document.createElement('select')).attr({
+      multiple: 'multiple',
```

Comment:
I know this is going to change a lot, but can you tell me the use case for picking multiple assets?

Reply:
I think this select box is just for adding assets - I image below this on the page there will be all the assets you've selected as relevant for the current disaster. This way you can mark multiple assets as relevant at a time. This select box won't be for any actually tagging/editing of how to use the elements.

Reply:
So these are assets that are not actually part of this disaster yet?

Reply:
correct - ideally once you make an asset part of a disaster it gets taken off that list. Currently these pickers are entirely state-based and don't have any affiliation with the actual disaster (yet)